internal/applause/applause.go

```diff
@@ -1 +1,278 @@
+	}
+
+	var inMeeting bool
+	for _, u := range meetingUserIDs {
+		if u == userID {
+			inMeeting = true
+			break
+		}
+	}
+
+	if !inMeeting {
+		return iccerror.NewMessageError(iccerror.ErrNotAllowed, "You are not part of meeting %d. Please be quiet.", meetingID)
+	}
+
+	if err := a.backend.ApplausePublish(meetingID, userID, time.Now().Unix()); err != nil {
+		return fmt.Errorf("publish applause in backend: %w", err)
+	}
+	return nil
+}
+
+// CanReceive returns an error, if the user can not receive applause.
+func (a *Applause) CanReceive(ctx context.Context, meetingID, userID int) error {
+	fetcher := datastore.NewFetcher(a.datastore)
+	if userID == 0 {
+		anonymousEnabled := fetcher.Field().Meeting_EnableAnonymous(ctx, meetingID)
+		if err := fetcher.Err(); err != nil {
+			return fmt.Errorf("fetching anonymous enabled: %w", err)
+
+		}
+		if !anonymousEnabled {
+			return iccerror.NewMessageError(iccerror.ErrNotAllowed, "Anonymous is not enabled")
+		}
+		return nil
+	}
+
+	meetingUserIDs := fetcher.Field().Meeting_UserIDs(ctx, meetingID)
+	if err := fetcher.Err(); err != nil {
+		return fmt.Errorf("fetching meeting users: %w", err)
+	}
+
+	var inMeeting bool
+	for _, u := range meetingUserIDs {
+		if u == userID {
+			inMeeting = true
+			break
+		}
+	}
+
+	if !inMeeting {
+		return iccerror.NewMessageError(iccerror.ErrNotAllowed, "You are not part of meeting %d.", meetingID)
+	}
+	return nil
+}
+
+// Receive returns the applause for a given meeting.
+func (a *Applause) Receive(ctx context.Context, tid uint64, meetingID int) (newTID uint64, msg MSG, err error) {
+	if tid == 0 {
+		present, err := a.presentUser(ctx, meetingID)
+		if err != nil {
+			return 0, MSG{}, fmt.Errorf("fetching present user: %w", err)
+		}
+		return a.topic.LastID(), MSG{0, present}, nil
+	}
+
+	for {
+		var messages []string
+		tid, messages, err = a.topic.Receive(ctx, tid)
+		if err != nil {
+			return 0, MSG{}, fmt.Errorf("receiving message from topic: %w", err)
+		}
+
+		// We are intressted in the last messaeg that has a entry for out
+		// meeting. We go backwards throw the messages and return, if we find
+		// something.
+		for i := len(messages) - 1; i >= 0; i-- {
+			var message map[int]MSG
+			if err := json.Unmarshal([]byte(messages[i]), &message); err != nil {
+				return 0, MSG{}, fmt.Errorf("decoding message from topic: %w", err)
+			}
+			if meetingData, ok := message[meetingID]; ok {
+				return tid, meetingData, nil
+			}
+		}
+	}
+}
+
+// LastID returns the newest id from the topic.
+func (a *Applause) LastID() uint64 {
+	return a.topic.LastID()
+}
+
+// Loop fetches the applause from the backend and saves it for the clients to
+// fetch.
+func (a *Applause) Loop(ctx context.Context, errHandler func(error)) {
+	if errHandler == nil {
+		errHandler = func(error) {}
+	}
+
+	lastApplause := make(map[int]int)
+
+	for {
+		if err := contextSleep(ctx, applauseInterval); err != nil {
+			return
+		}
+
+		d := time.Now().Add(-countTime)
```

Comment:
See above!

Reply:
The variable d is only used in two lines.